graphene/types/schema.py

```diff
@@ -54,6 +55,22 @@
 introspection_query = get_introspection_query()
 IntrospectionSchema = introspection_types["__Schema"]
 
+class UnforgivingExecutionContext(ExecutionContext):
+    """An execution context which doesn't swallow exceptions.
+    Instead it re-raises the original error, except for
+    GraphQLError, which is handled by graphql-core
+    """
+
+    def handle_field_error(
+        self,
+        error: GraphQLError,
+        return_type,
+    ) -> None:
+        if type(error.original_error) is GraphQLError:
```

Comment:
In our project, I had to change this to `isinstance(error.original_error, GraphQLError)`.

We have a few subclasses of `GraphQLError` to support transformations from different data types using `OurGraphQLError.from_other_error(...)`. This makes it easier to support `extension`'s within the errors.